### DISCUSSION TITLE
Diagram List Lay out Sub Node Order

### DISCUSSION THREAD

**Discussion by ThimotyS** [eclipse-sirius/sirius-components#3704] [Q&A] [answered]
Hello,

Currently I have a list lay-out for a certain model's sub nodes in my main model's diagram view. This is the view specification of said list lay-out:
![image](https://github.com/eclipse-sirius/sirius-web/assets/38472677/a0520baf-bf9f-42e2-a9a5-59f50359fb02)

When I instantiate the diagram view and look at the order of the list's sub nodes, then the order is different than the one seen in the tree view of the model. The order of sub nodes in the diagram view follows the definition order of my view specification instead of the order of the tree view.

In this example, the sub node da_train should be between sub nodes Action 1 and Action 3 in the diagram view, the same as in the tree view, but it isn't.
![image](https://github.com/eclipse-sirius/sirius-web/assets/38472677/3fb3695e-dda3-4485-afc8-8c4f119ace18)

Is there a way I can have the same order in my list from my tree to my diagram view? 

Kind regard,
T

**Comment by sbegaudeau** [comment, marked as answer]
Hi,

The order is indeed determined by the view specification on purpose. If you want to use the order of the semantic elements then you would have to use one child NodeDescription with one `semanticElementsExpression`.

Regards,

Stéphane Bégaudeau
